tests/phpunit/tests/test-customizer.php

```diff
@@ -0,0 +1,133 @@
+		require_once ABSPATH . WPINC . '/class-wp-customize-manager.php';
+	}
+
+	public function test_static_front_page_update() {
+		$_POST['customized'] = json_encode(
+			array(
+				'show_on_front' => 'page',
+				'page_on_front' => $this->page_fr,
+			)
+		);
+		$_POST['wp_customize'] = 'on';
+
+		$links_model             = self::$model->get_links_model();
+		$this->pll_env           = new PLL_Frontend( $links_model );
+		$GLOBALS['wp_customize'] = new WP_Customize_Manager();
+		$this->wp_customize      = $GLOBALS['wp_customize'];
+		do_action( 'customize_register', $this->wp_customize );
+		$this->pll_env->curlang = self::$model->get_language( 'fr' );
+		do_action( 'pll_language_defined', $this->pll_env->curlang->slug, $this->pll_env->curlang );
+
+		$show_on_front = $this->wp_customize->get_setting( 'show_on_front' );
+		$show_on_front->save();
+		$page_on_front = $this->wp_customize->get_setting( 'page_on_front' );
+		$page_on_front->save();
+
+		$this->assertSame( 'page', $show_on_front->value() );
+		$this->assertSame( $this->page_fr, $page_on_front->value() );
+	}
+
+	public function test_static_front_page_display_secondary_language() {
+		update_option( 'show_on_front', 'page' );
+		update_option( 'page_on_front', $this->page_en );
+		self::$model->clean_languages_cache();
+
+		$_POST['wp_customize'] = 'on';
+
+		$links_model             = self::$model->get_links_model();
+		$this->pll_env           = new PLL_Frontend( $links_model );
+		$GLOBALS['wp_customize'] = new WP_Customize_Manager();
+		$this->wp_customize      = $GLOBALS['wp_customize'];
+		do_action( 'customize_register', $this->wp_customize );
+		$this->pll_env->curlang = self::$model->get_language( 'fr' );
+		do_action( 'pll_language_defined', $this->pll_env->curlang->slug, $this->pll_env->curlang );
+
+		$show_on_front = $this->wp_customize->get_setting( 'show_on_front' );
+		$page_on_front = $this->wp_customize->get_setting( 'page_on_front' );
+
+		$this->assertSame( 'page', $show_on_front->value() );
+		$this->assertSame( $this->page_fr, $page_on_front->value() );
+	}
+
+	public function test_customize_registered_hooks_with_static_page_on_front() {
+		global $_wp_theme_features;
+
+		update_option( 'show_on_front', 'page' ); // Implicit `PLL_Frontend_Static_Pages` instance.
+		update_option( 'page_on_front', $this->page_en );
+		self::$model->clean_languages_cache();
+
+		// Switch to a block theme.
+		switch_theme( 'twentytwentythree' );
+		// Force the features.
+		$_wp_theme_features['block-templates']      = true;
+		$_wp_theme_features['block-template-parts'] = true;
+
+		$_POST['wp_customize'] = 'on';
+
+		$links_model   = self::$model->get_links_model();
+		$this->pll_env = new PLL_Frontend( $links_model );
+		$this->pll_env->init(); // Implicit `PLL_Frontend_Nav_Menu` instance.
+		$GLOBALS['wp_customize'] = new WP_Customize_Manager();
+		$this->wp_customize      = $GLOBALS['wp_customize'];
+		do_action( 'customize_register', $this->wp_customize );
```

Comment:
After discussing with @Hug0-Drelon: creating an instance of `WP_Customize_Manager` adds new hooks to `customize_register`, meaning that in that case we should not remove the menu item.

However, to make the test more clear about its intend, this can be simplified:
```suggestion
		add_action( 'customize_register', '__return_false' );
```